source/renderer/app/components/widgets/forms/PinCode.js

```diff
@@ -31,42 +39,258 @@ export default class PinCode extends Component<Props> {
+    const control: { blur?: Function, focus?: Function } = target;
+    const nextFieldFocusKey = inputKey + 1;
+    const nextInputField = this.inputsRef[nextFieldFocusKey];
+    // Recheck if input value is a separator value
+    const isSeparator = key === decimalSeparator || key === groupSeparator;
+    // Recheck if input value is a backspace value
+    const isBackSpace = key === 'Backspace';
+    // Get input field new value
+    const inputNewValue = this.inputsRef[inputKey]
+      ? this.inputsRef[inputKey].props.value
+      : null;
+    // Recheck if field is empty
+    const fieldIsEmpty = this.inputsRef[inputKey] ? !inputNewValue : false;
+    // Get cursor pointer position from input field
+    const selectionStart = this.inputsRef[inputKey]
+      ? this.inputsRef[inputKey].inputElement.current.selectionStart
+      : 0;
+    const selectionEnd = this.inputsRef[inputKey]
+      ? this.inputsRef[inputKey].inputElement.current.selectionEnd
+      : 0;
+    const isEntrySelected = selectionStart !== selectionEnd;
+    if (isSeparator) {
+      this.handleSeparatorInput(nextInputField, control);
+    }
+    this.handleBackspaceClick(
+      inputNewValue,
+      isBackSpace,
+      fieldIsEmpty,
+      inputKey,
+      isEntrySelected
+    );
+  };
+
+  handleBackspaceClick = (
+    inputNewValue: string | null,
+    isBackSpace: boolean,
+    fieldIsEmpty: boolean,
+    inputKey: number,
+    isEntrySelected: boolean
+  ) => {
+    const { value, onChange } = this.props;
+    const { focusKeyChanged } = this.state;
+    let focusKeyUpdated = false;
+    if (isBackSpace && fieldIsEmpty && !isEntrySelected) {
+      if (onChange) {
+        // Handle specific case when user pressed backspace and field is empty
+        // or cursor pointer position was in front of the value
+        if (this.isInputValueSelected(this.inputsRef[inputKey])) {
+          // Remove value from input field
+          value[inputKey] = '';
+        } else {
+          focusKeyUpdated = true;
+        }
+        this.setState({ isBackSpace, focusKeyChanged: focusKeyUpdated });
+        // Call onChange function to validate new value in focused input field
+        onChange(value, inputNewValue, inputKey);
+      }
+      this.focusKey = focusKeyChanged ? inputKey - 1 : inputKey;
+      this.add = false;
+      this.fromBackspace = true;
+    } else {
+      this.focusKey = inputKey;
+      this.fromBackspace = false;
+    }
+  };
+
+  setFocusOnField = (inputFieldRef: {
+    focus: ?Function,
+    props: ?Object,
+    inputElement: ?Object,
+  }) => {
+    const { focus, props, inputElement } = inputFieldRef;
+    if (focus) focus();
+    if (inputElement && props && props.value) {
+      inputElement.current.selectionStart = 0;
+      inputElement.current.selectionEnd = 1;
+    }
+  };
+
+  handleSeparatorInput = (
+    nextInputField: Field,
+    control: { blur?: Function, focus?: Function }
+  ) => {
+    if (nextInputField && nextInputField.focus) {
```

Comment:
if `nextInputField` it has focus for sure

Reply:
Done